File: src/types/communication.ts
export type ConversationType = 'DIRECT' | 'USER_PERSON' | 'GROUP';

export type MessageStatus = 'sent' | 'delivered' | 'read';

export interface PersonToPersonMessage {
  id: string;
  worldId: string;
  conversationId: string;
  senderPersonId: string;
  recipientPersonId: string;
  senderName?: string;
  senderAvatarEmoji?: string;
  content: string;
  createdAt: string;
  status: MessageStatus;
  metadata?: Record<string, unknown>;
}

export interface PersonConversation {
  id: string;
  worldId: string;
  type: ConversationType;
  participantIds: string[];
  topic?: string;
  lastMessage?: PersonToPersonMessage;
  lastActivityAt: string;
  exchangeCount: number;
  status: 'active' | 'completed' | 'paused';
  summary?: string;
  createdAt: string;
}

File: backend/app/schemas/communication.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any

class MessageBase(BaseModel):
    worldId: str
    conversationId: str
    senderPersonId: str
    recipientPersonId: str
    senderName: Optional[str] = None
    senderAvatarEmoji: Optional[str] = None
    content: str
    status: str = "sent"
    metadata: Optional[Dict[str, Any]] = None

class MessageCreate(MessageBase):
    pass

class MessageResponse(MessageBase):
    id: str
    createdAt: str

    class Config:
        from_attributes = True

class ConversationBase(BaseModel):
    worldId: str
    type: str = "DIRECT"
    participantIds: List[str]
    topic: Optional[str] = "Discussion"
    status: str = "active"
    summary: Optional[str] = None

class ConversationCreate(ConversationBase):
    pass

class ConversationResponse(ConversationBase):
    id: str
    lastActivityAt: str
    exchangeCount: int = 0
    createdAt: str
    lastMessage: Optional[MessageResponse] = None

    class Config:
        from_attributes = True
